src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -152,6 +152,7 @@ public IContent CreateContent(string name, int parentId, string contentTypeAlias
         {
             var contentType = FindContentTypeByAlias(contentTypeAlias);
             var content = new Content(name, parentId, contentType);
+            content.Path = string.Concat(content.Parent().IfNotNull(x => x.Path, content.ParentId.ToString()), ",", content.Id);

```

Comment:
in the core we should never ever use these 'helper' methods on services like Parent(). This will go back through the services layer using the ApplicationContext singleton!! :( This means that unit tests are way harder to write and since we're already inside the ContentService there's no reason to not just use it to find the parent instead of this syntactic sugar.